pkg/nsx/services/ipaddressallocation/ipaddressallocation_test.go

```diff
@@ -73,21 +75,41 @@ func createIPAddressAllocationService(t *testing.T) (*IPAddressAllocationService
 func Test_InitializeIPAddressAllocationStore(t *testing.T) {
 	service, mockController, _ := createService(t)
 	defer mockController.Finish()
+
+	// Init successfully
 	commonService := service.Service
 	patch := gomonkey.ApplyMethod(reflect.TypeOf(&commonService), "InitializeResourceStore", func(_ *common.Service, wg *sync.WaitGroup,
 		fatalErrors chan error, resourceTypeValue string, tags []model.Tag, store common.Store,
 	) {
 		wg.Done()
-		return
 	})
-	defer patch.Reset()
 
 	vpcService := &vpc.VPCService{}
 
 	_, err := InitializeIPAddressAllocation(commonService, vpcService, false)
 	if err != nil {
 		t.Error(err)
 	}
+
+	// Init failed
+	patch.Reset()
+	var callCount int32
+	patch = gomonkey.ApplyMethod(reflect.TypeOf(&commonService), "InitializeResourceStore", func(_ *common.Service, wg *sync.WaitGroup,
+		fatalErrors chan error, resourceTypeValue string, tags []model.Tag, store common.Store,
+	) {
+		defer wg.Done()
+		if atomic.AddInt32(&callCount, 1) > 1 {
+			fatalErrors <- errors.New("init failed")
+		} else {
+			return
+		}
+	})
+	defer patch.Reset()
+
+	_, err = InitializeIPAddressAllocation(commonService, vpcService, true)
+	if err != nil {
+		assert.Error(t, err)
```

Comment:
it better verify the error is init failed" or error message contains "init failed"